File: frontend/src/services/authService.js
import api from './api';

export const authService = {

  async login(email, password) {
    const response = await api.post('/auth/login', { email, password });
    return response.data.doctor;
  },


  async logout() {
    await api.post('/auth/logout');
  },

 
  async getCurrentDoctor() {
    const response = await api.get('/auth/me');
    return response.data.doctor;
  }
};

export default authService;

File: backend/services/authService.py
from backend.data.database import db


class AuthService:
    @staticmethod
    def authenticate_doctor(email, password):
        doctor = db.get_doctor_by_email(email)
        
        if not doctor:
            return None, "Doctor not found"
        
        if doctor['password'] != password:
            return None, "Invalid password"

        safe_doctor = {k: v for k, v in doctor.items() if k != 'password'}
        
        return safe_doctor, None

    @staticmethod
    def validate_session(doctor_id):
        return True

    @staticmethod
    def logout_doctor(doctor_id):
        return True